jwst/pipeline/calwebb_spec3.py

```diff
@@ -72,6 +75,16 @@ def process(self, input):
         output_file = input_models.meta.asn_table.products[0].name
         self.output_file = output_file
 
+        # Split the inputs into science and background
+        science_data, background_data = split_container(input_models)
+
+        # If background data are present, call the master background step
+        if len(background_data) > 0:
+            source_models = self.master_background(input_models)
+            source_models.meta.asn_table = input_models.meta.asn_table

```

Comment:
Do you want to copy the metadata from then entire `input_models`, or just from `science_data`?

Reply:
The problem that I was trying to solve is the fact that the `split_container` function, which is used here and within `master_background`, seemed to be returning models that did not have `asn_table` copied over, so it's not in either `science_data` or `source_models`, hence the only place left to get it from is the original `input_models`. Definitely needs further investigation. But you're right that for the purpose of the rest of the pipeline we only need or want the `asn_table` info pertaining to just the science images (not the backgrounds).